### ISSUE TITLE
target-colorspace-hint broken on new Nvidia Drivers

### ISSUE THREAD

**Issue by Asinin3** [mpv-player/mpv#15163]
### mpv Information

```bash
mpv v0.39.0-170-g37159a8e Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
 built on Oct 20 2024 00:05:18
libplacebo version: v7.349.0 (v7.349.0-18-g9e16c86-dirty)
FFmpeg version: N-117613-g72e538112
FFmpeg library versions:
   libavcodec      61.22.100
   libavdevice     61.4.100
   libavfilter     10.6.100
   libavformat     61.9.100
   libavutil       59.44.100
   libswresample   5.4.100
   libswscale      8.6.100
```

### Other Information

```text
- Windows version: W11 23H2
- GPU model, driver and version: RTX 4090, 566.03
- Source of mpv: https://github.com/shinchiro/mpv-winbuild-cmake/releases
- Introduced in version: N/A
```

### Reproduction Steps

Use Nvidia Driver 565.90 or later
Set "target-colorspace-hint=yes" in mpv.conf
Play a HDR video while windows is in SDR

### Expected Behavior

MPV Should set Windows to HDR with the above config when playing a HDR video

### Actual Behavior

HDR isn't being enabled automatically after the Nvidia driver update

### Log File

[output.txt](https://github.com/user-attachments/files/17489144/output.txt)

### I carefully read all instruction and confirm that I did the following:

- [X] I tested with the latest mpv version to validate that the issue is not already fixed.
- [X] I provided all required information including system and mpv version.
- [X] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [X] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [X] I attached the full, untruncated log file.
- [X] I attached the backtrace in the case of a crash.

**Comment by Asinin3** [comment]
Other reports of this:
https://www.reddit.com/r/mpv/comments/1fw8d9o/hdr_passthrough_stopped_working_after_the_latest/
https://www.reddit.com/r/mpv/comments/1g9s0wl/autoswitch_hdr_passthrough_still_not_working_with/

**Comment by llyyr** [comment]
Please report on Nvidia's forums or wherever they want you to report bugs. There's nothing mpv can do about this

**Comment by Asinin3** [comment]
> Please report on Nvidia's forums or wherever they want you to report bugs. There's nothing mpv can do about this

I understand, but its still good to have this tracked here for reference.



> Any way to bypass Windows HDR is a completely unsupported hack, so the "auto HDR toggle" is actually an unintended behavior. The [fix](https://developer.nvidia.com/vulkan-driver) for this bug has been merged into the R560 branch.
> 
> To enable HDR support, Windows HDR must be enabled. this is not NVIDIA or mpv's fault.
> 
> > August 7th, 2024 - Windows 552.93, Linux 550.40.67
> > New:
> > VK_EXT_device_fault
> > VK_NV_extended_sparse_address_space [Linux]
> > Fixes:
> > Improve Vulkan HDR behavior on Windows to not override user's Windows HDR Display settings

This is intriguing, many PC games on Windows could trigger enabling HDR in the OS, e.g Doom Eternal, MK11 or various Capcom games without the user manually changing their windows settings. I'll have to see if that broke on the new driver too then.

**Comment by laodaoxp** [comment]
> > 请在 Nvidia 论坛或他们希望您报告错误的任何地方进行报告。 mpv对此无能为力
> 
> 我明白，但在这里跟踪一下以供参考还是很好的。
> 
> > 任何绕过 Windows HDR 的方法都是完全不受支持的黑客行为，因此“自动 HDR 切换”实际上是一种意外行为。此错误[的修复](https://developer.nvidia.com/vulkan-driver)已合并到 R560 分支中。
> > 要启用 HDR 支持，必须启用 Windows HDR。这不是 NVIDIA 或 mpv 的错。
> > > 2024 年 8 月 7 日 - Windows 552.93、Linux 550.40.67
> > > 新增：
> > > VK_EXT_device_fault 
> > > VK_NV_extended_sparse_address_space [Linux]
> > > 修复：
> > > 改进 Windows 上的 Vulkan HDR 行为，以不覆盖用户的 Windows HDR 显示设置
> 
> 有趣的是，Windows 上的许多 PC 游戏都可以在操作系统中触发启用 HDR，例如《毁灭战士：永恒》、MK11 或各种 Capcom 游戏，而无需用户手动更改其 Windows 设置。我得看看新驱动程序是否也出现这种情况。

Is there a way to solve this problem? After reading the answer here Should be
2.August 7th, 2024 - Windows 552.93, Linux 550.40.67
Improve Vulkan HDR behavior on Windows to not override user's Windows HDR Display settings
My current solution is to use a driver lower than Nvidia Driver 565.90. For example: Driver version: 561.09
4. Currently, there is no other way to wait for me to come home from work. I am testing Potplayer
potplayer+LAV+MadVR
potplayer+LAV+MpcVideoRendere
This combination used to be able to automatically start HDR like MPVs, let's see if they are also affected?

**Comment by laodaoxp** [comment]
> 请在 Nvidia 论坛或他们希望您报告错误的任何地方进行报告。 mpv对此无能为力
I have already provided feedback and I am not sure if it has been effective
![000](https://github.com/user-attachments/assets/5218b156-c8a2-4904-9138-7eaf8f6334ee)

**Comment by kasper93** [comment]
> Set "target-colorspace-hint=yes" in mpv.conf
> Play a HDR video while windows is in SDR

No, it shouldn't. Vulkan/D3D11 graphic api are not suppose to switch the HDR mode set in Windows. This is the current policy.

There is no plans to add native HDR mode switching, same as we don't have resolution/frame-rate switching. You can use scripts for that.

**You can use Windows shortcut `Win+Alt+B` to toggle HDR mode on/off.** mpv now, after #16512 will react dynamically to the change and adjust the output as needed.

Let me know if there is something else we can do.
